Mathlib/Algebra/GroupWithZero/Range.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2025 Antoine Chambert-Loir and Filippo Nuccio. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, Filippo Nuccio
+-/
+
+import Mathlib.Algebra.Group.Submonoid.Defs
+import Mathlib.Algebra.GroupWithZero.Units.Basic
+import Mathlib.Tactic.NthRewrite
+
+/-! # The range of a `MonoidHomWithZero`,
+  when the codomain is a `GroupWithZero`
+
+If `f : A →* B` be a monoid hom, where `B` is a `CommGroupWithZero`,
+then `f.range₀` is the smallest submonoid of `B`
+containing the image of `f` which is a `CommGroupWithZero`.
+  carrier := { b | ∃ a c, f a ≠ 0 ∧  (f a * b = f c)}
+
+-/
+
+variable {A B : Type*} [MonoidWithZero A] [CommGroupWithZero B]
+  {F : Type*} [FunLike F A B] [MonoidHomClass F A B]
+  (f : F)
+
+namespace MonoidHomWithZero
+
+open Set
+
+/-- The range of a morphism of monoids with codomain a `CommGroupWithZero`,
```

Comment:
I was very confused by this definition until I realised that actually the codomain is a monoid with zero, not just a monoid. Maybe this should say something like "the group with zero generated by the range of a morphism from a monoid with zero to a group with zero"? or "the smallest group with zero containing the range of a morphism from a monoid with zero to a group with zero" or something?

Reply:
I agree, I was also a little confused by the docstring.

Also, you say "as a `CommGroupWithZero`" but here it is only a `Submonoid B`, so please mention that the actual instnace of `CommGroupWithZero` is added later.